kafka-client/src/main/java/dev/responsive/kafka/config/ResponsiveConfig.java

```diff
@@ -190,4 +211,57 @@ public String toString() {
   public ResponsiveConfig(final Map<?, ?> originals) {
     super(CONFIG_DEF, originals);
   }
+
+  public SubPartitioner getSubPartitioner(
+      final CassandraClient cassandraClient,
+      final Admin admin,
+      final TableName name,
+      final String changelogTopicName
+  ) {
+    final OptionalInt actualRemoteCount = cassandraClient.numPartitions(name.cassandraName());
+    final int kafkaPartitions;
+    try {
+      kafkaPartitions = admin.describeTopics(List.of(changelogTopicName))
+          .allTopicNames()
+          .get()
+          .get(changelogTopicName)
+          .partitions()
+          .size();
+    } catch (InterruptedException | ExecutionException e) {
+      throw new RuntimeException(e);
+    }
+
+    return getSubPartitioner(
+        actualRemoteCount,
+        kafkaPartitions,
+        getInt(STORAGE_DESIRED_NUM_PARTITION_CONFIG),
+        name,
+        changelogTopicName
+    );
+  }
+
+  static SubPartitioner getSubPartitioner(
+      final OptionalInt actualRemoteCount,
+      final int kafkaPartitions,
+      final int desiredNum,
+      final TableName name,
+      final String changelogTopicName
+  ) {
+    final int factor = (desiredNum == NO_SUBPARTITIONS) ? 1 : desiredNum / kafkaPartitions;
+    final int computedRemoteNum = factor * kafkaPartitions;
+
+    if (actualRemoteCount.isPresent() && actualRemoteCount.getAsInt() != computedRemoteNum) {
+      throw new IllegalArgumentException(String.format("%s was configured to %d, which "
+              + "given %s partitions in kafka topic %s would result in %d remote partitions "
+              + "for table %s (remote partitions must be a multiple of the kafka partitions). "
+              + "The remote store is already initialized with %d partitions - it is backwards "
+              + "incompatible to change this. Please set %s to %d.",
+          STORAGE_DESIRED_NUM_PARTITION_CONFIG, desiredNum, kafkaPartitions,
+          changelogTopicName, computedRemoteNum, name.cassandraName(),
```

Comment:
super nit: prefer the kafkaName in any user-facing error/logs
```suggestion
          changelogTopicName, computedRemoteNum, name.kafkaName(),
```